core/domain/user_jobs_one_off.py

```diff
@@ -95,27 +101,42 @@ def entity_classes_to_map_over(cls):
     def map(item):
         """Implements the map function for this job."""
 
-        # Assumption: We do not have any profile user in the UserSettingsModel,
-        # because we have no method of associating profile users with a full
-        # user.
+        # Assumptions:
+        # 1) We are only having full users in the database, and no
+        #    profile users. For profiles we need to do a role check and ensure
+        #    the gae_id attribute is None for them.
+        # 2) This migration works correctly only if gae_id cannot be updated
+        #   in any way for a user. If a gae_id is mapped to some other user by
+        #   deleing it, an error will be raised while migration (see below).
```

Comment:
Wording: either ``during migration`` or ``while migrating``

Reply:
Changed the description now as @seanlip suggested in [this comment](https://github.com/oppia/oppia/pull/10123#discussion_r467410927)